app/services/clients/connector.client.js

```diff
@@ -301,11 +307,40 @@ ConnectorClient.prototype = {
   },
 
   /**
- *
- * @param {Object} params
- * @param {Function} successCallback
- * @returns {Promise}
- */
+   * Checks Worldpay 3DS Flex credentials
+   *
+   * @param {Object} params
+   * @param {Function} successCallback
+   * @returns {Promise}
+   */
+  postCheckWorldpay3dsFlexCredentials: function (params) {
+    return new Promise((resolve, reject) => {
+      const url = _getCheckWorldpay3dsFlexCredentialsUrlFor(params.gatewayAccountId, this.connectorUrl)
+      const startTime = new Date()
+      const context = {
+        description: 'Check Worldpay 3DS Flex credentials',
+        method: 'POST',
+        service: 'connector',
+        url: url,
+        defer: { resolve: resolve, reject: reject },
+        startTime: startTime,
+        correlationId: params.correlationId,
+        body: params.payload
+      }
+      requestLogger.logRequestStart(context)
+
+      const callbackToPromiseConverter = createCallbackToPromiseConverter(context)
+      oldBaseClient.post(url, params, callbackToPromiseConverter)
```

Comment:
👋  this review is just for JavaScript rather than functionality, it would be ideal if this could use the new "baseClient" as we're trying to incrementally phase out this older pattern. We're doing that because: 
* creating promises ourselves higher in the stack and passing `resolve`, `reject` around can lead to tight coupling and hard to follow code
* the pub/ sub method of async used for error handling requires a new instance of the client on every request which has caused issues in the past

The interface for the new base client is very similar! There are plenty of examples in the `connector.client.js` file, it might look something like: 
```js 
postCheckWorldpay3dsFlexCredentials: function postCheckWorldpay3dsFlexCredentials(params) {
  return baseClient.post({
    baseUrl: this.connectorUrl,
    url: CHECK_WORLDPAY_3DS_FLEX_CREDENTIALS_PATH.replace('{accountId}', params.gatewayAccountId),
    // ...
  })
}
```

The current client is already promise based underneath so just returning that will accomplish the same thing. 

Please let me know if this isn't clear or if I can help out making this change!

Reply:
Thanks @sfount. I will make the changes.

Reply:
I have updated to use `baseClient.post()`.